posts/2024-10-03-introducing-grails-6-2-1.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Introducing Grails Framework 6.2.1 Release
+date: October 3, 2024
+description: It highlights the changes and improvements made in this release, provides information about bug fixes, dependency upgrades, and acknowledges the contributions of various developers.
+author: James Fredley
+image: grails-blog-index-3.png
+---
+
+# [%title]
+
+[%author]
+
+[%date]
+
+The Grails Foundation is excited to announce the release of Grails framework 6.2.1. This latest version delivers critical bug fixes and reintroduces key features based on community feedback.
+
+## What's Changed
+
+The 6.2.1 release, while being a patch release, brings back the Grails Shell, Profiles and Wrapper (as an optional feature) and addresses some bugs.
+
+* **Grails Shell, Profiles and Wrapper**: 
+    * Reintroduces the Grails Shell and Profiles in response to community feedback, addressing a key feature loss identified by users including:
+      * enables the IntelliJ Grails Plugin to load in IntelliJ  
+      * execution of Grails scripts that have not been migrated to Gradle such as **grails list-plugins** and **grails dbm-update**
+      * execution of custom company/internal Grails scripts
+      * continued use of custom company/internal profiles
+    * Grails Wrapper is now an optional feature from Grails Forge via the Grails distribution or [https://start.grails.org/](https://start.grails.org/])
+      * you can now run **./grailsw update-wrapper** to get the latest grails wrapper version based on the Grails version
+
+## Bug Fixes and Improvements
+
+Grails 6.2.1 also addresses several bug fixes and improvements, including:
+
+* Fixes a lack of a “controller” attribute in GSP tags with a “resource” attribute
+* Fixes a bug in LinkGenerator when domain classes have a static id mapping to another property, that is not “id”
+* Fixes a problem finding the main class when running CLI commands
```

Comment:
It also: "Fixes an edge cases that causes a cast class exception when the same variable name is used as arguments to multiple actions"

Reply:
https://github.com/grails/grails-core/issues/13486

Reply:
Updated